src/JobPanel/backend/json_data.py

```diff
@@ -0,0 +1,176 @@
+    """
+    def __init__(self, config):
+        """
+        Initialize class dict from config serialization.
+        :param config:
+        """
+        for k, v in config.items():
+            if k in self.__dict__:
+                # dict
+                if isinstance(self.__dict__[k], dict):
+                    if isinstance(v, dict):
+                        for k2, v2 in v.items():
+                            if k2 in self.__dict__[k]:
+                                obj = self._simple(self.__dict__[k][k2], v2)
+                                if obj is not None:
+                                    self.__dict__[k][k2] = obj
+                # list
+                elif isinstance(self.__dict__[k], list):
+                    if len(self.__dict__[k]) == 0:
+                        if isinstance(v, list):
+                            self.__dict__[k] = v
+                    elif len(self.__dict__[k]) == 1:
+                        if isinstance(v, list):
+                            l = []
+                            for vi in v:
+                                obj = self._simple(self.__dict__[k][0], vi)
+                                if obj is not None:
+                                    l.append(obj)
+                            self.__dict__[k] = l
+                # tuple
+                elif isinstance(self.__dict__[k], tuple):
+                    if isinstance(v, list) and len(self.__dict__[k]) == len(v):
+                        l = []
+                        for ki, vi in zip(self.__dict__[k], v):
+                            obj = self._simple(ki, vi)
+                            if obj is not None:
+                                l.append(obj)
+                        self.__dict__[k] = tuple(l)
+                else:
+                    obj = self._simple(self.__dict__[k], v)
+                    if obj is not None:
+                        self.__dict__[k] = obj
+
+    def _simple(self, temp, data):
+        # JsonData
+        if isinstance(temp, JsonData):
+            return temp.__class__(data)
+        # ClassFactory
+        elif isinstance(temp, ClassFactory):
+            return temp.make_instance(data)
+        # str
+        elif isinstance(temp, str):
+            if isinstance(data, str):
+                return data
+        # int
+        elif isinstance(temp, int):
+            if isinstance(data, int):
+                return data
+        # float
+        elif isinstance(temp, float):
+            if isinstance(data, float) or isinstance(data, int):
+                return float(data)
+        # bool
+        elif isinstance(temp, bool):
+            if isinstance(data, bool):
+                return data
+        # other
+        else:
+            return data
+        return None
+
+    def serialize(self):
+        """
+        Serialize the object into JSON.
+        :return:
+        """
+        return json.dumps(self._get_dict(), sort_keys=True)
```

Comment:
Serialization must return just dictionaries not JSON yet. So, we can put result in other envelopes and make true serialization just before message is send.